### PR TITLE
Fix/icd slicing

### PR DESCRIPTION
**Pull request by noemide**
Removed still existing ValueSet binding on value[x] Level.
Reworked slicing completly since i ran into an Issue where non-ICD codes were matched to more than one slice resulting in an error.
(This time, i validated against the updated examples too!)

### CODE REVIEW

File: extension/Diagnosis.StructureDefinition.json [hunk 1]
```diff
@@ -25,15 +25,12 @@
...
+        "min": 1,
         "type": [
           {
             "code": "CodeableConcept"
           }
-        ],
-        "binding": {
-          "strength": "required",
-          "valueSet": "https://fhir.bbmri.de/ValueSet/SampleContentDiagnosis"
-        }
```
Comment (noemide) [inline extension/Diagnosis.StructureDefinition.json:36]: Forgot to remove this in last version...

File: extension/Diagnosis.StructureDefinition.json [hunk 2]
```diff
@@ -47,25 +44,34 @@
...
+            },
+            {
+              "type": "value",
+              "path": "$this"
             }
           ],
-          "rules": "open"
-        }
+          "rules": "closed"
+        },
```
Comment (noemide) [inline extension/Diagnosis.StructureDefinition.json:54]: Make sure only slices with ICD or our codes exist

File: extension/Diagnosis.StructureDefinition.json [hunk 3]
```diff
@@ -47,25 +44,34 @@
...
       },
       {
         "id": "Extension.value[x].coding:icd-10-who",
         "path": "Extension.value[x].coding",
-        "sliceName": "icd-10-who",
-        "binding": {
-          "strength": "required",
-          "description": "Sample Content in ICD-10 WHO or unspecific",
-          "valueSet": "https://fhir.bbmri.de/ValueSet/SampleContentDiagnosisICD-10-WHO"
-        }
```
Comment (noemide) [inline extension/Diagnosis.StructureDefinition.json:63]: Removed binding, now done via fixed URI (same in other ICD slices)

File: extension/Diagnosis.StructureDefinition.json [hunk 4]
```diff
@@ -47,25 +44,35 @@
             {
               "type": "value",
               "path": "version"
+            },
+            {
+              "type": "value",
+              "path": "$this"
```
Comment (noemide) [inline extension/Diagnosis.StructureDefinition.json:50]: Necessary since last slice (no-icd) is bound to value set and does not have fixed URI

**Comment by alexanderkiel** [comment]
I really can't decide if this solves our problem. Can you just add the relevant examples here so that we see that the validator does the right thing?

**Review by alexanderkiel — APPROVED** [review]

**Pull request merged**